catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -131,9 +136,17 @@ private void queryNext(int index) {
                 Map<String, Serializable> props = new HashMap<>();
                 // Avoid caching all results while dumping with native query mode
                 props.put("mode", "native");
-                response = catalog.query(filter.apply(index));
+                response = catalog.query(new QueryRequestImpl(new QueryImpl(filter,
+                        pageIndex,
+                        pageSize,
```

Comment:
Good question. `-1` used to mean get everything and was actually caped by some sources and endpoints when necessary. So it could technically cause some OOM errors and such when interfacing with endpoints that do not limit that number (e.g., CSW), but that's already a problem today.

Having said that, we could use a constant number here to prevent potential problems. The flip side though is that if the system property used to cap the number of metacards is set higher than this number, say on a system that stores small metacards, then this code will not take full advantage of that maximum page size.

I guess as long as we make this number high enough and don't expose it to the clients of this class, I'm ok either way.